scripts/version/README.md

```diff
@@ -0,0 +1,54 @@
+
+## 1. How version is set in *.jar artifacts
+
+### 1.1 Semantic version
+The java artifacts' semantic version is updated by the [`set-version.ps1`](./set-version.ps1). The script updates the following files:
+1.  [root pom.xml](../../pom.xml)
+2.  [sonar-csharp-plugin pom.xml](../../sonar-csharp-plugin/pom.xml)
+3.  [sonar-vbnet-plugin pom.xml](../../sonar-vbnet-plugin//pom.xml)
+4.  [sonar-dotnet-shared-library pom.xml](../../sonar-dotnet-shared-library/pom.xml)
+5.  [its/pom.xml](../../its/pom.xml) (this project is not related to the artifacts, but the root pom is its parent pom and if there's no match the project build fails)
+
+### 1.2 Build metadata
+The `buildNumber` is updated in the sonar-dotnet [Azure pipeline](https://github.com/SonarSource/sonar-dotnet/blob/a998f32eac72c7b6b4562935ffb8d423c6ebf936/azure-pipelines.yml#L504) is and equal to the pipeline [`buildId`](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/run-number?view=azure-devops&tabs=yaml#tokens). More specifically the pipeline is calling the [`update-maven-version-steps.yml`](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_git/pipelines-yaml-templates?path=/update-maven-version-steps.yml&version=GBmaster&line=16&lineEnd=17&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contentsl), which in turn calls the [`compute-build-version-step.yml`](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_git/pipelines-yaml-templates?path=/compute-build-version-step.yml&version=GBmaster&line=29&lineEnd=30&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) that attaches the `buildNumber` to the semantic version.
+
+## 2. How is the version set for the for the NuGet and the *.dll artifacts
+
+### 2.1 Semantic version
+
+The semantic version for these files is updated also by the [`set-version.ps1`](./set-version.ps1). The files updated are:
+1. [Version.props](./Version.props)
+1. [SonarAnalyzer.CSharp.nuspec](../../analyzers/packaging/SonarAnalyzer.CSharp.nuspec)
+1. [SonarAnalyzer.VisualBasic.nuspec](../../analyzers/packaging/SonarAnalyzer.VisualBasic.nuspec)
+1. [SonarAnalyzer.CFG.cs.nuspec](../../analyzers/src/SonarAnalyzer.CFG/SonarAnalyzer.CFG.cs.nuspec)
+1. [SonarAnalyzer.CSharp.Styling.nuspec](../../analyzers/packaging/SonarAnalyzer.CSharp.Styling.nuspec)
+1. [AssemblyInfo.Shared.cs](../../analyzers/src/AssemblyInfo.Shared.cs)
+
+However, during the sonar-dotnet pipeline run, the files 2-6 above, have any reference to the semantic version replaced by the value in the [MainVersion field](https://github.com/SonarSource/sonar-dotnet/blob/a998f32eac72c7b6b4562935ffb8d423c6ebf936/scripts/version/Version.props#L3) of the `Version.props` file. This happens during the build of the [`ChangeVersion.proj`](./ChangeVersion.proj) that is [executed in the pipeline](https://github.com/SonarSource/sonar-dotnet/blob/master/azure-pipelines.yml#L91).
```

Comment:
The pipeline #L91 needs a fixed commit. Otherwise it will go out of sync with very next pipeline change